canister/src/rpc_client/json/mod.rs

```diff
@@ -163,6 +163,43 @@ pub struct GetBlockConfig {
     pub max_supported_transaction_version: Option<u8>,
 }
 
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(into = "(Vec<String>, Option<GetSignatureStatusesConfig>)")]
+pub struct GetSignatureStatusesParams(Vec<String>, Option<GetSignatureStatusesConfig>);
+
+impl GetSignatureStatusesParams {
+    pub fn num_signatures(&self) -> usize {
+        self.0.len()
+    }
+}
+
+impl From<sol_rpc_types::GetSignatureStatusesParams> for GetSignatureStatusesParams {
+    fn from(params: sol_rpc_types::GetSignatureStatusesParams) -> Self {
+        let config = if params.is_default_config() {
+            None
+        } else {
+            Some(GetSignatureStatusesConfig {
+                search_transaction_history: params.search_transaction_history,
+            })
+        };
+        Self(params.signatures, config)
+    }
+}
+
+impl From<GetSignatureStatusesParams> for (Vec<String>, Option<GetSignatureStatusesConfig>) {
+    fn from(params: GetSignatureStatusesParams) -> Self {
+        (params.0, params.1)
+    }
+}
+
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct GetSignatureStatusesConfig {
+    #[serde(rename = "searchTransactionHistory")]
+    pub search_transaction_history: Option<bool>,
```

Comment:
why the additional option here, given that the top-level type already contains `Option<GetSignatureStatusesConfig>`?

```suggestion
    pub search_transaction_history: bool,
```

Reply:
This has the slight disadvantage of us having to define a default value for this config field. Though this is only theoretical since in practice the field should always be `Some(..)` if the config is not default.

Reply:
> This has the slight disadvantage of us having to define a default value for this config field.

I don't get it, you could just do `search_transaction_history.map(GetSignatureStatusesConfig)` like the suggestion in this [comment](https://github.com/dfinity/sol-rpc-canister/pull/96#discussion_r2084768361)?

Reply:
You are right yes, this is actually cleaner. Done.